dashboard/app/models/reference_guide.rb

```diff
@@ -62,6 +62,21 @@ def remove_serialization
     File.delete(file_path)
   end
 
+  def copy_to_course_version(course_version)
+    copied_ref_guide = ReferenceGuide.find_or_create_by(
+      course_version_id: course_version.id,
+      key: key
+    ) do |created_ref_guide|
+      created_ref_guide.update!(
+        parent_reference_guide_key: parent_reference_guide_key,
```

Comment:
What happens if there isn't a guide with this key in the new course version?

Reply:
this shouldn't happen without manual intervention. when a reference guide is deleted, it deletes the whole tree below it, and same with moving an existing guide. the `parent_reference_guide_key` is a loose association, so the order of creation shouldn't matter as long as they all get copied. if we do end up in that state somehow, the guides with parents that don't exist will just be hidden (see `organize_reference_guides` in referenceGuideHelpers.js) and would require some eng work to find and clean up.